docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,525 @@
+```
+
+#### Update implementation of existing plugin functions
+These plugin functions include `_verify_evidence()`, `_verify_report()`
+and they both invoke function `oe_verify_quote_with_sgx_endorsements()`
+for quote verification.
+
+- Update API `oe_verify_quote_with_sgx_endorsements()` implementation as below:
+  - Try to call DCAP QVL to verify quote first
+    - In host-side plugin:
+      - Call host-side function `oe_sgx_verify_quote()` with SGX endorsements,
+        which will call SGX QVL library to verify quote.
+    - In enclave-side plugin:
+      - Construct structure `sgx_ql_qe_report_info_t` as below, `nonce` and
+        `app_enclave_target_info` are input, qve_report is QvE report which
+        target to application enclave.
+      ```C
+      typedef struct _sgx_ql_qe_report_info_t {
+        sgx_quote_nonce_t nonce;
+        sgx_target_info_t app_enclave_target_info;
+        sgx_report_t qve_report;
+      }sgx_ql_qe_report_info_t;
+      ```
+      - Call OCALL function `oe_verify_quote_ocall()` to host-side plugin library
+        to verify SGX quote.
+      - With all the QvE return values, call new API.
+        `oe_verify_qve_report_and_identity()` (described below) to
+        verify QvE report and identity
+  - If DCAP QVL returns error to indicate that QVL doesn't not exist on current
+    platform, then automatically fallback to the existing implementation
+  - Add log information to indicate whether the QVL-based or existing implementation
+    is used
+- If QVL-based quote verification is used, in order to exactly align with current
+  implementation, only SGX ECDSA quote with `UpToDate` TCB status would be accepted
+  in step 1.
+
+
+#### Add function in enclave-side plugin for QvE Identity verification
+For enclave-side verifier plugin, after quote verification, it needs to verify that the result
+comes from a trusted QvE.
+
+Intel® SGX SDK provides a library named `sgx_dcap_tvl` to help the verifier to
+verify QvE's identity, you can refer to [source file](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteVerification/dcap_tvl/sgx_dcap_tvl.cpp)
+
+This library uses hardcoded QvE identity values, because:
+- There is no need for x509 and JSON parser in verifier enclave's TCB.
+- Most of QvE's identity attributes only change infrequently.
+
+The only identity info may change more often is QvE's ISV SVN, so the verifier needs
+to provide a SVN number as threshold, only when current QvE's ISV SVN is equal to
+or larger than this threshold, the verifier can trust the QvE verification result.
+The verifier can get latest QvE ISV SVN from [QvE Identity at Intel® Provisioning
+Cert Server](https://api.portal.trustedservices.intel.com/documentation#pcs-qve-identity-v2)
+
+As OE SDK cannot use Intel® SGX SDK trusted library directly, we need to port
+this library into OE SDK as part of enclave-side plugin.
+
+```C
+oe_result_t oe_verify_qve_report_and_identity(
```

Comment:
To confirm: are you proposing this as an _internal_ API?

Reply:
> To confirm: are you proposing this as an _internal_ API?

Yes, this is an internal API used by verifier plugin to verify QvE identity.